xbmc/platform/linux/OptionalsReg.cpp

```diff
@@ -57,6 +57,23 @@ bool OPTIONALS::PulseAudioRegister()
 }
 #endif
 
+//-----------------------------------------------------------------------------
+// OSS
+//-----------------------------------------------------------------------------
+#ifdef TARGET_FREEBSD
```

Comment:
platform specific optionals need to go to platform specific optionalsreg

Reply:
Right now there is no FreeBSD-specific OptionalsReg.cpp. Do you want me to replicate `xbmc/platform/linux/OptionalsReg.cpp` for FreeBSD?

I could also introduce an HAS_OSS or something if you like that better.

I could also just remove TARGET_FREEBSD because AFAICT AESinkOSS will build on Linux too. But it's an open question if you want that given that it has been disabled on Linux in 525af3efe2acafb729f767061fe8489a52f57f1b.

Reply:
in order to keep it here in linux folder, it has to work on Linux, not just compile. if this is the case, HAS_OSS is ok.

otherwise you have to create a folder xbmc/platform/freebsd

Reply:
Right. So I'm not going to run test this on Linux, so that option is a no go.

> otherwise you have to create a folder xbmc/platform/freebsd

Doesn't that imply replicating a bunch of files from `xbmc/platform/linux` just for FreeBSD (where they will probably start to rot soon enough)? It would also mean that I have to add `#ifdef TARGET_FREEBSD` things to files in `xbmc/windowing` that currently include `platform/linux/OptionalsReg.h`. This seems like a big overcomplication.